testing/upload_validation.ts

```diff
@@ -227,6 +228,44 @@ export function validateSyncTableSchema(
   );
 }
 
+/**
+ * Returns a map of sync table names to their parent sync table names, or undefined if the hierarchy is invalid.
+ * Example valid return: { Child: 'Parent' }
```

Comment:
are we going to use this function in the workflow? if we would like to create child ingestion from parent one, is it better to return `{Parent: 'Child'}` instead?

Reply:
also thinking through what's needed during the child ingestion creation, we might also need to know how to build up the execution parameters for child ingestion. Would you have any thoughts on where to store the data? should it be in this mapping or we store it in the schema somewhere? 

The parameters could be `{parentPropertyParam: rowData[parentPropertyKey], grandParentPropertyParam: parentParameters[grandParentPropertyParam]}`

Reply:
We should be able to parse the necessary data out of the childKey, right @johnli-codaio ?

Reply:
Yeah I can include these params inside the childKey, but wouldn't these execution params also just be passed into the `ingestionParam` column in our `csb_ingestion_configs` table?

We would need the SDK to expose the execution params somehow for it's children (ex: repoUrl for pull requests, calendarId for events, etc.) I think?

Reply:
To me I thought the childKey was mostly for identification and deduping purposes, not necessarily for holding ingestion params, but I could use it for ingestion params as well (at the very least, enough to identify the entity we're crawling, like pullRequestId or calendarId, etc.)

Reply:
exactly, we need to store `parentPropertyParam`, `parentPropertyKey`, and maybe grandparent ones later to build the `ingestionParam` column. Where do we generate `childKey`?

Reply:
I'm thinking we can generate childKey in the `process_csb_ingestion` workflow, before we insert a new ingestion and upsert a new config. You can add a TODO for me to handle that.

Reply:
But `ingestionParam` also needs be built in the same place in order to upsert child ingestion? 
<img width="694" alt="image" src="https://github.com/coda/packs-sdk/assets/102989081/8fca0cc5-cf3a-4719-9c0c-505d93018029">

Reply:
I'm thinking what we do is:

1. get Sync Hierarchy
2. Calculate childKey based on syncHierarchy and some identification of the entity we are crawling (repoUrl, calendarId, etc.)
3. Build ingestionParameters and all config stuff 
4. We might need a new SQL query (insert_child_ingestion) or something like that that would be a TXN that would:
  a. Insert the child ingestion first. If it conflicts (a childKey already exists) then we just return. Keeps the SQL idempotent
  b. Insert the child ingestion config with the ingestionParams.

Reply:
Yeah, this looks good to me, just would like to find an easy way to get `parentPropertyParam` and `parentPropertyKey` in the SDK, current function only returns function names mapping

Reply:
I think it might make sense to double-store the data in the child key and the ingestion params, though I'm not sure.

But to answer Evan, sorry I misunderstood you before, I was imagining something like:
```
const currentPageOfSyncResults = [...]
const childTables = getSyncTableHierarchy(pack)[currentTable];
for (const childTable of childTables) {
  const syncTableDef = pack.syncTables.find(...);
  const crawlDefinedParams = syncTableDef.getter.parameters.filter(p=>p.crawlStrategy...);
  for (const parentRow of currentPageOfSyncResults) {
    const crawlParams = {};
    for (const paramDef of crawlDefinedParams) {
      crawlParams[paramDef.name] = parentRow[paramDef.crawlStrategy.parentTable.propertyKey];
    }
    // create key
    // insert ingestion/config
  }
}
```

Reply:
Oh, got it, that makes sense. We would have to double store if it's required by dedupe. Thanks!